src/Message/MessageParser.ts

```diff
@@ -15,16 +15,18 @@ export default function parseMessage(
 
 	let command: string | undefined;
 	const params: MessageParam[] = [];
-	let tags: Map<string, string> | undefined;
+	let tags: Map<string, string> = new Map();
```

Comment:
This breaks the check in line 23 and thus prevents tags from ever being parsed.

The default value should be assigned after the loop, if the variable is still undefined at that point.

Reply:
You've got a good eye. That should do the trick; The whole tags stuff is beyond the scope of the original PR.

I have enabled "Allow edits from maintainers".